scipy/sparse/linalg/_dsolve/tests/test_linsolve.py

```diff
@@ -722,6 +722,17 @@ class TestSpsolveTriangular:
     def setup_method(self):
         use_solver(useUmfpack=False)
 
+    def test_zero_diagonal(self):
+        n = 5
+        A = np.random.randn(n, n)
+        b = np.arange(n)
+        A = scipy.sparse.tril(A, k=0, format='csr')
+
+        x = spsolve_triangular(A, b, unit_diagonal=True, lower=True)
+
+        A.setdiag(1)
+        assert_array_almost_equal(A.dot(x), b)
```

Comment:
We should not use anymore `assert_array_almost_equal` (cf NumPy's doc)
```suggestion
        assert_allclose(A.dot(x), b)
```